calliope/test/test_constraint_results.py

```diff
@@ -441,6 +440,62 @@ def test_demand_share_per_timestep_equals(self):
         # assert share in each timestep is 0.6
         assert ((expensive_generation / demand).round(5) == 0.6).all()
 
+    def test_demand_share_per_timestep_decision_inf(self):
```

Comment:
Another comment: should there be a test or two for:
1. using a conversion technology which consumes electricity
2. transmission techs with an efficiency < 100%
3. multiple carriers in the model, inc. a share of both/one being set as a decision variable